### PR TITLE
Refactor FXIOS-7576 [v121] Replace link buttons in FakespotOptInCardView

### PR DESCRIPTION
**Pull request by imsanchit**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7576)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16863)

## :bulb: Description
**Screenshots before the PR**
Screenshot 2023-11-01 at 7 23 17 PM
Screenshot 2023-11-01 at 7 24 01 PM
Screenshot 2023-11-02 at 4 55 33 PM
Screenshot 2023-11-02 at 4 55 46 PM

**Screenshots after the PR**
Screenshot 2023-11-03 at 2 56 48 PM
Screenshot 2023-11-03 at 2 57 00 PM
Screenshot 2023-11-03 at 2 57 11 PM
Screenshot 2023-11-03 at 2 56 34 PM

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift [hunk 1]
```diff
@@ -8,27 +8,23 @@ import UIKit
...
-                                         bottom: UX.buttonVerticalInset,
-                                         right: UX.buttonHorizontalInset)
     }
 
     public func configure(viewModel: LinkButtonViewModel) {
         accessibilityIdentifier = viewModel.a11yIdentifier
         setTitle(viewModel.title, for: .normal)
         titleLabel?.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .body,
                                                                   size: viewModel.fontSize)
+        contentEdgeInsets = viewModel.contentEdgeInsets
```
Comment (lmarceau) [inline BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift:25]: We have some warnings related to `contentEdgeInsets`, could we look into how to fix this at the same time 🤔 ?

File: BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift [hunk 2]
```diff
@@ -8,27 +8,23 @@ import UIKit
...
         titleLabel?.textAlignment = viewModel.textAlignment
+        contentHorizontalAlignment = viewModel.contentHorizontalAlignment
```
Comment (lmarceau) [inline BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift:27]: Do we need `titleLabel?.textAlignment` if we have `contentHorizontalAlignment`?

Comment (imsanchit) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift:27]: I think it can be removed, shall I go ahead and make this change?

Comment (lmarceau) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift:27]: If we get same result, then I would say yes

Comment (imsanchit) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift]: yes, the results are the same and so updated the PR.

File: BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift (hunk 1 above)
Comment (imsanchit) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LinkButton.swift]: Hey @lmarceau,  I have used UIButton.Configuration to fix this, please have a look.

File: BrowserKit/Sources/ComponentLibrary/Buttons/LinkButtonViewModel.swift [hunk 3]
```diff
@@ -10,15 +10,18 @@ public struct LinkButtonViewModel {
...
     public let fontSize: CGFloat
-    public let textAlignment: NSTextAlignment
+    public let contentInsets: NSDirectionalEdgeInsets
+    public let contentHorizontalAlignment: UIControl.ContentHorizontalAlignment
 
     public init(title: String,
                 a11yIdentifier: String,
                 fontSize: CGFloat = 16,
-                textAlignment: NSTextAlignment = .left) {
+                contentInsets: NSDirectionalEdgeInsets = .zero,
```
Comment (lmarceau) [inline BrowserKit/Sources/ComponentLibrary/Buttons/LinkButtonViewModel.swift:19]: Could we have a default `contentInsets` that isn't zero? Maybe we can take the one from `OnboardingCard` as the default for now.

Comment (imsanchit) [reply BrowserKit/Sources/ComponentLibrary/Buttons/LinkButtonViewModel.swift]: Hey, have updated the PR with the same, please have a look.

**Comment by lmarceau** [comment]
@imsanchit it seems the build failed due to new warnings, can we figure out which ones are new? Just trying to determine if we should increase the build warnings threshold due to the `.iOS(.v14)` -> `.iOS(.v15)` change

**Comment by lmarceau** [comment]
Ok, I had a look and it seems the increase warning count is because of the version bump in BrowserKit. I created tasks to fix the warnings [here](https://github.com/mozilla-mobile/firefox-ios/issues?q=is%3Aissue+is%3Aopen+label%3Awarnings), so to unblock this PR we can increase the warnings threshold from 92 to 104 under this [file](https://github.com/mozilla-mobile/firefox-ios/blob/1bbc2d0f459f93e066f3bc34a6212f1e9d2efd2b/test-fixtures/generate-metrics.sh#L7-L8) in the meantime they are fixed 🙏 Thank you!

**Comment by imsanchit** [comment]
Thanks @lmarceau for pointing this out, have updated the threshold warning count. Please try if this succeeds.

**Review by lmarceau — APPROVED** [review]

**Comment by thatswinnie** [comment]
@Mergifyio backport release/v120

**Pull request merged**